src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,391 @@
+    data is accepted.
+
+    Parameters
+    ----------
+    sampling_values
+        Values where sample data has been evaluated or the full-order model should be evaluated.
+        Sampling values are represented as a nested list `svs` such that `svs[i]`corresponds
+        to sampling values of the `i-th` variable.
+    samples_or_fom
+        Can be either a full-order model (|TransferFunction| or |Model| with a `transfer_function`
+        attribute) or data sampled at the values specified in `sampling_values`. Samples are
+        represented as a tensor `S`. E.g., for 3 inputs `S[i,j,k]` corresponds to the sampled
+        value at `(sampling_values[0][i],sampling_values[1][j],sampling_values[2][k])`. In the
+        MIMO case `S[i,j,k]` represents a matrix of dimension `dim_output` times `dim_input`.
+    conjugate
+        Whether to compute complex conjugates of first sampling variables and enforce
+        interpolation in complex conjugate pairs (allows for constructing real system matrices).
+    nsp_tol
+        Tolerance for null space of higher-dimensional Loewner matrix to check for
+        interpolation or convergence.
+    post_process
+        Whether to do post-processing or not. If the Loewner matrix has a null space
+        of dimension greater than 1, it is assumed that the algorithm converged to
+        a non-minimal order interpolant which may cause numerical issues. In this case,
+        the post-processing procedure computes an interpolant of minimal order.
+    L_rk_tol
+        Tolerance for ranks of 1-D Loewner matrices computed in post-processing.
+    """
+
+    def __init__(self, sampling_values, samples_or_fom, conjugate=True, nsp_tol=1e-16, post_process=True,
+                 L_rk_tol=1e-8):
+        if isinstance(samples_or_fom, TransferFunction) or hasattr(samples_or_fom, 'transfer_function'):
+            fom = samples_or_fom
+            if not isinstance(samples_or_fom, TransferFunction):
+                fom = fom.transfer_function
+            self.num_vars = 1 + len(fom.parameters)
+
+            assert len(sampling_values) == self.num_vars
+            self.parameters = fom.parameters
+            self.samples = np.empty([len(sv) for sv in sampling_values] + [fom.dim_output, fom.dim_input],
+                                    dtype=sampling_values[0].dtype)
+            for idx, vals in zip(np.ndindex(self.samples.shape[:-2]),
+                                 itertools.product(*sampling_values)):
+                params = {p: v for p, v in zip(fom.parameters, vals[1:])}
+                self.samples[idx] = fom.eval_tf(vals[0], mu=params)
+            if fom.dim_input == fom.dim_output == 1:
+                self.samples = self.samples.reshape(self.samples.shape[:-2])
+        else:
+            self.samples = samples_or_fom
+            self.num_vars = len(sampling_values)
+            self.parameters = {f'p{i}': 1 for i in range(self.num_vars-1)}
```

Comment:
This could actually just be `{'p': self.num_vars}`, or `{'s': 1, 'p': self.num_vars - 1}` for linear time-dependent problems.

Reply:
Since only `TransferFunction` models are supported it should just be `{'p': self.num_vars - 1}`, right? `s` does not count as a parameter for the ROM.

Reply:
Right, true.